x-pack/filebeat/module/salesforce/login/ingest/pipeline.yml

```diff
@@ -0,0 +1,368 @@
+---
+description: Pipeline for parsing salesforce login logs
+processors:
+- set:
+    field: event.ingested
+    value: '{{_ingest.timestamp}}'
+- remove:
+    field:
+      - message
+    ignore_missing: true
+
+- set:
+    field: salesforce.access_mode
+    value: rest
+    if: ctx?.json?.TIMESTAMP_DERIVED != null
+    ignore_failure: true
+- set:
+    field: salesforce.access_mode
+    value: stream
+    if: ctx?.json?.EventDate != null
+    ignore_failure: true
+
+- date:
+    field: json.TIMESTAMP
+    target_field: "@timestamp"
+    formats: ["yyyyMMddHHmmss.SSS"]
+    ignore_failure: true
+- date:
+    field: json.EventDate
+    target_field: "@timestamp"
+    formats:
+    - ISO8601
+    ignore_failure: true
+
+- rename:
+    field: json.API_TYPE
+    target_field: salesforce.login.api_type
+    ignore_missing: true
+- rename:
+    field: json.ApiType
+    target_field: salesforce.login.api_type
+    ignore_missing: true
+- rename:
+    field: json.AUTHENTICATION_METHOD_REFERENCE
+    target_field: salesforce.login.auth_service_id
+    ignore_missing: true

```

Comment:
Is this renaming correct? json.AUTHENTICATION_METHOD_REFERENCE into salesforce.login.auth_service_id?

Reply:
My bad. I'll update it, thank you!